kratos/input_output/stl_io.cpp

```diff
@@ -208,16 +234,23 @@ void StlIO::ReadLoop(ModelPart & rThisModelPart)
     ReadKeyword("loop");
 
     *mpInputStream >> word; // Reading vertex or endloop
-    std::size_t node_id = block_for_each<MaxReduction<std::size_t>>(
-        rThisModelPart.GetRootModelPart().Nodes(),
-        [](NodeType& rNode) { return rNode.Id();}) + 1;
+
     Element::NodesArrayType temp_geom_nodes;
     while(word == "vertex"){
         Point coordinates = ReadPoint();
-        temp_geom_nodes.push_back(rThisModelPart.CreateNewNode(node_id++, coordinates[0], coordinates[1], coordinates[2] ));
+        temp_geom_nodes.push_back(rThisModelPart.CreateNewNode(mNextNodeId++, coordinates[0], coordinates[1], coordinates[2] ));
         *mpInputStream >> word; // Reading vertex or endloop
     }
-    rThisModelPart.CreateNewGeometry("Triangle3D3", temp_geom_nodes);
+    const std::string new_entity_type = mParameters["new_entity_type"].GetString();
```

Comment:
this is quite inefficient, since it is parsed for every triangle

I would instead create a lambda or std::function once, which you can pass

Reply:
Ok! done ;)